hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -389,6 +392,19 @@ async def delete_object(self, bucket: str, name: str, **kwargs) -> None:
     async def list_objects(self, bucket: str, **kwargs) -> PageIterator:
         return PageIterator(self, f'/b/{bucket}/o', kwargs)
 
+    async def compose(self, bucket: str, names: List[str], destination: str, **kwargs) -> None:
+        n = len(names)
+        if n == 0:
+            raise ValueError('no components in compose')
+        if n > 32:
+            raise ValueError(f'too many components in compose, maximum of 32: {n}')
+        assert 'json' not in kwargs
+        assert 'body' not in kwargs
```

Comment:
body and json both set the body of the request, just in different formats.  If json is given, it is serialized to create the body.  aiohttp doesn't allow both of them to be set.  I could let aiohttp generate the error, but I thought that it would be cleaner to check it here since they don't set json, we do.